Emby.Naming/Video/VideoListResolver.cs

```diff
@@ -21,27 +21,39 @@ public IEnumerable<VideoInfo> Resolve(List<FileSystemMetadata> files, bool suppo
         {
             var videoResolver = new VideoResolver(_options);
 
-            var videoInfos = files
-                .Select(i => videoResolver.Resolve(i.FullName, i.IsDirectory))
-                .Where(i => i != null)
-                .ToList();
+            List<VideoFileInfo> videoInfos = new List<VideoFileInfo>();
+            foreach (var file in files)
+            {
+                var tmp = videoResolver.Resolve(file.FullName, file.IsDirectory);
+                if (tmp != null)
+                {
+                    videoInfos.Add(tmp);
+                }
+            }
 
             // Filter out all extras, otherwise they could cause stacks to not be resolved
             // See the unit test TestStackedWithTrailer
-            var nonExtras = videoInfos
-                .Where(i => string.IsNullOrEmpty(i.ExtraType))
-                .Select(i => new FileSystemMetadata
+            IEnumerable<FileSystemMetadata> GetNonExtras(IEnumerable<VideoFileInfo> videoFiles)
```

Comment:
can you explain why change this?

Reply:
I find find the longer `.Select`s harder to read.
Don't know what you guys prefer.